python/ray/tune/utils/file_transfer.py

```diff
@@ -0,0 +1,234 @@
+import io
+import os
+import shutil
+import tarfile
+
+from typing import Optional, Tuple, Dict, Generator, Union
+
+import ray
+
+
+_DEFAULT_CHUNK_SIZE_BYTES = 500 * 1024 * 1024
+
+
+def sync_dir_between_nodes(
+    source_ip: str,
+    source_path: str,
+    target_ip: str,
+    target_path: str,
+    force_all: bool = False,
+    chunk_size: int = _DEFAULT_CHUNK_SIZE_BYTES,
+    _return_all_remotes: bool = False,
+) -> Union[ray.ObjectRef, Tuple[ray.ObjectRef, ray.ActorID, ray.ObjectRef]]:
+    """Synchronize directory on source node to directory on target node.
```

Comment:
Can we beef up that this is only about kicking off the sync/scheduling the sync rather than really having it synced by the time the function returns?

Reply:
I've just updated the public functions, it defaults to blocking calls and only returns futures when `return_futures=True`